src/compiler/core.ts

```diff
@@ -1937,8 +1937,10 @@ namespace ts {
         return compareValues(a?.start, b?.start) || compareValues(a?.length, b?.length);
     }
 
-    export function min<T>(a: T, b: T, compare: Comparer<T>): T {
-        return compare(a, b) === Comparison.LessThan ? a : b;
+    export function min<T>(items: readonly [T, ...T[]], compare: Comparer<T>): T;
+    export function min<T>(items: readonly T[], compare: Comparer<T>): T | undefined;
+    export function min<T>(items: readonly T[], compare: Comparer<T>): T | undefined {
+        return reduceLeft(items, (x, y) => compare(x, y) === Comparison.LessThan ? x : y);

```

Comment:
Wouldn't it have been better to switch the call-site to

```ts
reduceLeft(items, min)
```

leaving the old definition alone?

Reply:
Oh, I guess you would have needed to pass in a comparer.

```ts
reduceLeft(items, (x, y) => min(x, y, compareMatches));
```

Reply:
Okay, never mind, there are more call sites than just two now. I just have no idea how often `betterMatch` gets called, so I feel like it might just be better to have two helper functions.

Reply:
IMO it’s silly for `min` to take only two items to compare. It had one call site before; now it has two. At first I preserved the two-argument form and made an overload for the array form, but the function implementation with the combined overload signature was disgusting. I considered inlining the `reduce` call into the call site, but `min` is so much more readable. If anything, the old implementation could be inlined into `betterMatch`. Fewer, better utils seems better to me.